### DISCUSSION TITLE
Why doesn't any query option take effect?

### DISCUSSION THREAD

**Discussion by sawa-ko** [mikro-orm/mikro-orm#2969] [Q&A]
Note that for example, the fields option does nothing and neither does the size option, all fields are always displayed and the multi-item search ignores the size and always gets all items.

I am using mikro-orm with nestjs.

![image](https://user-images.githubusercontent.com/56084970/160670526-8d48eff6-2835-4b42-9fa0-e3f337110381.png)

![image](https://user-images.githubusercontent.com/56084970/160670845-9fd14115-5cb4-4e9f-82c8-3e955284219e.png)

**Comment by sawa-ko** [comment]
@B4nan you can help? sorry about the ping.

**Reply by B4nan** [reply]
Not sure what you expect from `limit: 0`, the first one is probably an entity already loaded in the context, try doing `em.fork().find()` or add `refresh: true` to the find options.

**Reply by sawa-ko** [reply]
@B4nan Thanks for your answer. 🙌

 in the first case it happens every time, no matter if I restart the server, and try to load that entity for the first time.

Also, does adding the "refresh" option have any impact on server and/or database resource usage?

In the second case I am doing a pagination of items and it seems strange to me not to be able to put 0 as a value.

**Reply by B4nan** [reply]
Well, you can put 0 in there, and it will be used as you can see in the query log, so `.limit(0)`, the question is why that does not have any effect on the results. Seems more like error in the mongo connector, the ORM code seems to be correct.

https://github.com/mikro-orm/mikro-orm/blob/master/packages/mongodb/src/MongoConnection.ts#L138-L141

So does `refresh: true` help with the first thing or not? If yes, it really means you got this entity already loaded in your context. You can as well check what is loaded via `em.getUnitOfWork().getIdentityMap()`.

**Reply by sawa-ko** [reply]
Same result, I don't know what is happening. 😿
My mikro-orm configuration is as follows:
```ts
// app.module.ts
// imports
  MikroOrmModule.forRootAsync({
      imports: [ConfigModule],
      inject: [ConfigService],
      useFactory: () => ({
        type: 'mongo',
        autoLoadEntities: true,
        strict: true,
        validate: true,
        ensureIndexes: true,
        highlighter: new MongoHighlighter(),
        entities: [...EmbeddedUserEntities, ...EmbeddedMessageEntities],
        validateRequired: true,
        logger(message) {
          new Logger('Database').log(message);
        },
        cache: {
          enabled: true,
          pretty: true,
          adapter: MemoryCacheAdapter,
        },
        resultCache: {
          adapter: MemoryCacheAdapter,
          expiration: 2e3,
        },
      }),
    }),

// auth.module.ts
// imports
MikroOrmModule.forFeature({ entities: [UserEntity, UserAuthSessionsEntity] })

// auth.service.ts
// constructor
@InjectRepository(UserEntity)
private readonly _userRepository: EntityRepository<UserEntity>,

// mikro-orm packages versions info
{
  "@mikro-orm/core": "^5.1.1",
  "@mikro-orm/mongodb": "^5.1.1",
  "@mikro-orm/nestjs": "^5.0.1"
}

// mongodb server version
// MongoDB 5.0.5 Community
```

![image](https://user-images.githubusercontent.com/56084970/160842844-67b4279c-481a-4ed9-afbd-f80cb239861c.png)

**Reply by sawa-ko** [reply]
@B4nan any ideas? :(
